charts/envoy/templates/deployment.yaml

```diff
@@ -82,36 +82,37 @@ spec:
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
           volumeMounts:
-            {{- if .Values.tls.upstream.caRootCertSecret }}
-            - name: scalar-envoy-tls-ca-root-volume
-              mountPath: /etc/envoy/upstream/tls/ca.pem
-              subPath: ca-root-cert
-            {{- end }}
-            {{- if .Values.tls.downstream.certChainSecret }}
-            - name: scalar-envoy-tls-cert-chain-volume
-              mountPath: /etc/envoy/cert.pem
-              subPath: cert-chain
+            {{- if .Values.tls.upstream.enabled }}
+            - name: scalar-envoy-tls-upstream-volume
+              mountPath: /etc/envoy/upstream/tls
             {{- end }}
-            {{- if .Values.tls.downstream.privateKeySecret }}
-            - name: scalar-envoy-tls-private-key-volume
-              mountPath: /etc/envoy/key.pem
-              subPath: private-key
+            {{- if .Values.tls.downstream.enabled }}
+            - name: scalar-envoy-tls-downstream-volume
+              mountPath: /etc/envoy/downstream/tls
             {{- end }}
       volumes:
-        {{- if .Values.tls.upstream.caRootCertSecret }}
-        - name: scalar-envoy-tls-ca-root-volume
+        {{- if .Values.tls.upstream.enabled }}
+        - name: scalar-envoy-tls-upstream-volume
           secret:
+            {{- if .Values.tls.upstream.caRootCertSecret }}
             secretName: {{ .Values.tls.upstream.caRootCertSecret }}
+            {{- else }}
+            secretName: {{ include "envoy.fullname" . }}-envoy-tls-cert
+            {{- end }}
```

Comment:
If you set `tls.upstream.caRootCertSecret` explicitly, this chart uses the specified root CA certificate. However, if you don't specify that parameter, this chart uses the root CA certificate for downstream TLS.

In other words, if you use the same CA to issue certificate for Envoy and upstream Scalar products, you don't need to set root CA certificate manually.

You can see the conditions on the document side.
https://github.com/scalar-labs/docs-internal-orchestration/pull/24/files#r1584029281